### PR TITLE
cleanup json crate

### PR DESCRIPTION
**Pull request by niklasad1**
I got annoyed that some of the types had documentation that said only for tests which weren't the case. So I did a cleanup, sorry for a big PR

It does the following:
- converts json to `rust 2018`
- adds feature `test-helpers`
- moves all tests helpers to `test-helpers mod`
- removes duplicated types
- some style fixes

### CODE REVIEW

File: ethcore/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ ethabi = "8.0"
 ethabi-contract = "8.0"
 ethabi-derive = "8.0"
 ethash = { path = "../ethash", optional = true }
-ethjson = { path = "../json", optional = true }
+ethjson = { path = "../json", features = ["test-helpers"], optional = true }
```
Comment (niklasad1) [inline ethcore/Cargo.toml]: nit, fix

File: ethcore/pod/src/account.rs [hunk 2]
```diff
@@ -85,22 +85,6 @@ impl PodAccount {
 	}
 }
 
-impl From<ethjson::blockchain::Account> for PodAccount {
```
[the remark this answers is not in the document]
Comment (niklasad1) [reply ethcore/pod/src/account.rs]: good question, we had two similar structures that were more or less redundant and `ethjson::blockchain::{State, Account}` were only used for tests

For example we had these two accounts types:

```rustc
pub struct BlockchainAccount {
	/// Balance.
	pub balance: Uint,
	/// Code.
	pub code: Bytes,
	/// Nonce.
	pub nonce: Uint,
	/// Storage.
	pub storage: BTreeMap<Uint, Uint>,
	/// Version.
	#[serde(default)]
	pub version: Uint,
}

pub struct SpecAccount {
	/// Builtin contract.
	pub builtin: Option<Builtin>,
	/// Balance.
	pub balance: Option<Uint>,
	/// Nonce.
	pub nonce: Option<Uint>,
	/// Code.
	pub code: Option<Bytes>,
	/// Version.
	pub version: Option<Uint>,
	/// Storage.
	pub storage: Option<BTreeMap<Uint, Uint>>,
	/// Constructor.
	pub constructor: Option<Bytes>,
}
```

As you can see `Spec::Account` had the same fields as `Option` with some extra but if they are missing they will just be deserialized to `None` so I think it is fine but ofc it required a few changes in the JSON-tests

File: json/src/transaction.rs [hunk 3]
```diff
@@ -14,14 +14,12 @@
...
 
-use uint::Uint;
-use bytes::Bytes;
-use hash::Address;
-use maybe::MaybeEmpty;
+use crate::{bytes::Bytes, hash::{Address, H256}, maybe::MaybeEmpty, uint::Uint};
+use serde::Deserialize;
 
-/// Transaction test transaction deserialization.
+/// Plain transaction deserialization.
```
[the remark this answers is not in the document]
Comment (niklasad1) [reply json/src/transaction.rs]: yeah, that was bad lemme clarify. Basically I meant a transaction without any signing info, will change!

File: ethcore/src/json_tests/difficulty.rs [hunk 4]
```diff
@@ -21,9 +21,13 @@ use spec::Spec;
...
 
-pub fn json_difficulty_test<H: FnMut(&str, HookType)>(json_data: &[u8], spec: Spec, start_stop_hook: &mut H) -> Vec<String> {
+pub fn json_difficulty_test<H: FnMut(&str, HookType)>(
+	json_data: &[u8],
+	spec: Spec,
+	start_stop_hook: &mut H
+) -> Vec<String> {
 	let _ = ::env_logger::try_init();
-	let tests = ethjson::test::DifficultyTest::load(json_data).unwrap();
+	let tests = ethjson::test_helpers::difficulty::DifficultyTest::load(json_data).unwrap();
```
[the remark this answers is not in the document]
Comment (niklasad1) [reply ethcore/src/json_tests/difficulty.rs]: let's do an explicit import here instead, it is still clear to me.

File: ethcore/pod/src/account.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvdplm) [reply ethcore/pod/src/account.rs]: Good catch. Paying down technical debt. :)

File: json/src/test_helpers/transaction.rs [hunk 5]
```diff
@@ -17,16 +17,17 @@
...
+
+/// Type for running `Transaction` tests
+pub type Test = super::tester::GenericTester<String, TransactionTest>;
 
 /// Transaction test deserialization.
 #[derive(Debug, Deserialize)]
 pub struct TransactionTest {
 	pub rlp: Bytes,
-	pub _info: ::serde::de::IgnoredAny,
+	pub _info: serde::de::IgnoredAny,
```
Comment (dvdplm) [inline json/src/test_helpers/transaction.rs]: Need docs or annotation.

File: json/src/test_helpers/transaction.rs [hunk 6]
```diff
@@ -17,16 +17,17 @@
...
 	#[serde(flatten)]
 	pub post_state: BTreeMap<ForkSpec, PostState>,
```
Comment (dvdplm) [inline json/src/test_helpers/transaction.rs]: missing docs

**Pull request merged**